lib/internal/cluster/master.js

```diff
@@ -106,25 +115,77 @@ function setupSettingsNT(settings) {
 }
 
 function createWorkerProcess(id, env) {
-  var workerEnv = util._extend({}, process.env);
-  var execArgv = cluster.settings.execArgv.slice();
-  var debugPort = 0;
+  // If master process started with debug, workers shoud use ports with offset
+  // This code parses execArgv from cluster.settings
+  // and augments values of ports in following flags:
+  // --inspect, --inspect-port, --debug, --debug-brk, --debug-port
+
+  const workerEnv = util._extend({}, process.env);
+  const execArgv = cluster.settings.execArgv.slice();
+  const ipv6Regex = /^(\[[\d:]+\])(?::(.+))?$/;
+  let ipv6Match;
+  let debugSocket;
+  let splittedDebugSocket;
+  let debugPort = 0;
+  let debugHost = '';
+  let splittedDebugArg;
+  let resultArg;
 
   util._extend(workerEnv, env);
   workerEnv.NODE_UNIQUE_ID = '' + id;
 
-  for (var i = 0; i < execArgv.length; i++) {
-    const match = execArgv[i].match(
-      /^(--inspect|--inspect-(brk|port)|--debug|--debug-(brk|port))(=\d+)?$/
-    );
+  // Cycle is reversed on purpose:
+  // node currently selects debug port from last argument
+  // so we're changing last argument and breaking the cycle
+  let i = execArgv.length;
+  while (i--) {
+    if (debugArgRegex.test(execArgv[i])) {
+      splittedDebugArg = execArgv[i].split('=');
```

Comment:
Grammar nit: 'splitted' is not a word (anymore - I've seen it used in old texts), use 'split'.

Style: I'd use a destructuring assignment, ex. `let [arg, port] = execArgv[i].split('=');`